proxy/proxy.go

```diff
@@ -533,6 +539,29 @@ func callWeaveAttach(container *docker.Container, args []string) error {
 	return nil
 }
 
+func validateCIDRs(cidrs []string) error {
+	prefixes := []string{"ip:", "net:", ""}
+
+	for _, cidr := range cidrs {
+		if cidr == "net:default" {
+			continue
+		}
+		ok := false
+		for _, prefix := range prefixes {
+			if strings.HasPrefix(cidr, prefix) {
+				if _, err := address.ParseCIDR(strings.TrimPrefix(cidr, prefix)); err == nil {
+					ok = true
+				}
+				break
+			}
+		}
+		if !ok {
+			return fmt.Errorf("invalid CIDR: %s", cidr)
```

Comment:
Because of the "" prefix the 'if' inside the loop will succeed at least once. So why not simply return error/nil there? Then we don't need all that 'ok' logic.